extensions/wikia/AdEngine/js/AmazonMatch.js

```diff
@@ -96,16 +115,34 @@ define('ext.wikia.adEngine.amazonMatch', [
 		return amazonCalled;
 	}
 
-	function getPageParams() {
-		log(['getPageParams', targetingParams], 'debug', logGroup);
-		return {
-			amznslots: targetingParams
-		};
+	function getSlotParams(slotName) {
+		log(['getSlotParams'], 'debug', logGroup);
+
+		var ret = {};
+
+		// No Object.keys on IE8
+		if (!Object.keys) {
+			log(['filterSlots()', 'no Object.keys (IE8?)'], 'error', logGroup);
+			return {};
```

Comment:
What do you think about having just one `return` in function? Here it'd just require to change `if` body with what's below, add `else` and only log the error there - the value of `ret` will be returned and it will be empty in case of IE8.